pydantic_settings/sources.py

```diff
@@ -459,7 +479,9 @@ def prepare_field_value(self, field_name: str, field: FieldInfo, value: Any, val
         """
         is_complex, allow_parse_failure = self._field_is_complex(field)
         if is_complex or value_is_complex:
-            if value is None:
+            if isinstance(value, EnvNoneType):
```

Comment:
This was a missing case I found from https://github.com/pydantic/pydantic-settings/pull/204, setting a complex field to `None`.

Reply:
Also noted that I did not add an explicit test case. Will add.

Reply:
Added test cases.